src/social-network-provider/twitter.com/ui/fetch.ts

```diff
@@ -0,0 +1,155 @@
+import {
+    bioCard,
+    fromPostSelectorsSelectPostContentString,
+    postPopupSelector,
+    postsContentSelectors,
+    postsRootSelector,
+    postsSelectors,
+    selfInfoSelectors,
+} from '../utils/selector'
+import { MutationObserverWatcher } from '@holoflows/kit'
+import Services from '../../../extension/service'
+import { PersonIdentifier } from '../../../database/type'
+import { host } from '../index'
+import { getEmptyPostInfo, SocialNetworkUI } from '../../../social-network/ui'
+import { deconstructPayload } from '../../../utils/type-transform/Payload'
+import { regexMatch, timeout } from '../../../utils/utils'
+import { hasPostPopup } from '../utils/status'
+import { equal, notEmpty, notNullable } from '../../../utils/assert'
+
+export const resolveLastRecognizedIdentity = (self: SocialNetworkUI) => {
+    const selfSelector = selfInfoSelectors().screenName
+    const assign = () => {
+        try {
+            const ref = self.lastRecognizedIdentity
+            const info = selfInfoSelectors().screenName.evaluate()
+            const id = new PersonIdentifier(host, notEmpty(info, 'user id not found')[0])
+            equal(id.isUnknown, false, 'user id not recognized')
+            ref.value = { identifier: id }
+        } catch (e) {
+            /**
+             * This function expects to fail to collect info,
+             * maybe don't throw is a good idea
+             *                                  -- Jack-Works
+             */
+            console.log(e)
+        }
+    }
+    new MutationObserverWatcher(selfSelector)
+        .enableSingleMode()
+        .addListener('onAdd', () => assign())
+        .addListener('onChange', () => assign())
+        .startWatch()
+        .then()
+}
+
+export const resolveInfoFromBioCard = () => {
```

Comment:
We should call it resolveInfoFromBioCardTwitter to avoid accidentally import at someone else.